File: output/deepseek_chat/avatar/direct_translation/python/java/atcoder_ABC150_D.java

import java.util.*;
import java.math.*;

public class atcoder_ABC150_D {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int num = sc.nextInt();
        Set<Integer> numSet = new HashSet<>();
        for (int i = 0; i < n; i++) {
            numSet.add(sc.nextInt());
        }
        Set<Integer> twoTimesSet = new HashSet<>();
        boolean flag = true;
        for (int i : numSet) {
            for (int j = 1; j < 30; j++) {
                i /= 2;
                if (i % 2 != 0) {
                    twoTimesSet.add(j);
                    break;
                }
            }
            if (twoTimesSet.size() != 1) {
                System.out.println(0);
                flag = false;
                break;
            }
        }
        if (flag) {
            List<Integer> numList = new ArrayList<>(numSet);
            int lcm = numList.get(0);
            for (int i = 1; i < numList.size(); i++) {
                lcm = lcm * numList.get(i) / gcd(lcm, numList.get(i));
            }
            System.out.println((num - lcm / 2) / lcm + 1);
        }
    }

    private static int gcd(int a, int b) {
        return b == 0 ? a : gcd(b, a % b);
    }
}

File: output/deepseek_r1/avatar/direct_translation/java/python/atcoder_ABC150_D.py

import math

n, m = map(int, input().split())
a = [int(x) // 2 for x in input().split()]

lcm = 1
for num in a:
    g = math.gcd(lcm, num)
    lcm = (lcm * num) // g
    if lcm > m:
        print(0)
        exit()

for num in a:
    if (lcm // num) % 2 == 0:
        print(0)
        exit()

count = m // lcm
print((count + 1) // 2)

